pkgs/applications/virtualization/xen/patches.nix

```diff
@@ -0,0 +1,63 @@
+# Patching Xen? Check the XSAs at
+# https://xenbits.xen.org/xsa/
+# and try applying all the ones
+# we haven't gotten around to
+# yet, if any are necessary.
+
+{ fetchpatch }:
+
+let
+  xsaPatch =
+    { name, hash }:
+    (fetchpatch {
+      url = "https://xenbits.xen.org/xsa/xsa${name}.patch";
+      inherit hash;
+    });
+  qubesPatch =
+    {
+      name,
+      tag,
+      hash,
+    }:
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/QubesOS/qubes-vmm-xen/v${tag}/${name}.patch";
+      inherit hash;
+    });
+in
+{
+  # Example patches:
+  #
+  # "XSA_100" = xsaPatch {
+  #   name = "100";
+  #   hash = "sha256-0000000000000000000000000000000000000000000000000000";
+  # };
+  #
+  # "QUBES_libxl-fix-all-issues" = qubesPatch {
+  #   name = "1000-libxl-fix-all-issues";
+  #   tag = "4.20.0-1";
+  #   hash = "sha256-0000000000000000000000000000000000000000000000000000";
+  # };
+
+  # Build reproducibility patches for Xen 4.17.4, but works on 4.18.2.
```

Comment:
```suggestion
  # Build reproducibility patches for Xen 4.17.4, but Qubes did not yet updated them and they work on 4.18.2.
```

Reply:
Agreed. Added to staging.

Reply:
cross-ref https://github.com/NixOS/nixpkgs/pull/324693#discussion_r1685830233